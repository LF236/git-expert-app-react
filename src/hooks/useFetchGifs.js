import { useState, useEffect } from "react";
import getGif from "../helpers/getGifs";

const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        getGif(category)
            .then(imgs => {
                setTimeout(() => {
                    setState({
                        data: imgs,
                        loading: false
                    })
                }, 0);                
            })
    }, [category]);

    return state;
}

export default useFetchGifs;