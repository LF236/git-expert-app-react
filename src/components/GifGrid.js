import React from 'react'
import PropTypes from 'prop-types';
import useFetchGifs from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';

const GifGrid = ({ category }) => {
    const { data: images, loading } = useFetchGifs( category );

    return (
        <>
            <h2 className="animate__animated animate__bounceIn">{ category }</h2>
            { loading && <p className="animate__animated animate__flash">Loading...</p> }
            <div className="gif-grid animate__animated animate__fadeIn">
                {
                    images.map(img => {
                        return <GifGridItem key={ img.id } { ...img }/>
                    })
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid;