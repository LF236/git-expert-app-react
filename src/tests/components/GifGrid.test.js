import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';
jest.mock( '../../hooks/useFetchGifs' );
describe( 'Pruebas sobre el componente <GifGrid />', () => {
    let category = 'naruto';
    test( 'El componente renderizado debe ser igual al Snapshot por defecto', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category }/> );
        
        expect( wrapper ).toMatchSnapshot();
    } );

    test( 'Debe mostrar items cuando se cargar imágenen con useFetchGifs', () => {
        let imgs = [{
            id: 'dasdsa',
            url: 'https://localhost/nada/img.png',
            title: 'Luis Fernando Rodriguez Hernández'
        }];

        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={ category }/> );
        // expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'p' ).exists() ).toBe( false );
        expect( wrapper.find( 'GifGridItem' ).length ).toBe( imgs.length );
    } )
} )