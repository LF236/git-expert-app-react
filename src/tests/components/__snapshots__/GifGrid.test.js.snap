// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GifGrid /> El componente renderizado debe ser igual al Snapshot por defecto 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__bounceIn"
  >
    naruto
  </h2>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="gif-grid animate__animated animate__fadeIn"
  />
</Fragment>
`;
